app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt

```diff
@@ -122,7 +135,7 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
     }
 
-    private fun setGuestRoomLinkFeatureFlag(userId: UserId) {
+    private fun setGuestRoomLinkFeatureFlag(userId: UserId) =
```

Comment:
Maybe these functions could just be `suspend`, and not call `viewModelScope.launch` ?

This way, the caller has the responsibility of managing async things, decomposing work, etc.